certbot_dns_cpanel/authenticator.py

```diff
@@ -155,17 +187,18 @@ def _get_zone_and_name(self, record_domain):
         data = self.data.copy()
         data['cpanel_jsonapi_func'] = 'fetchzones'
 
+        logger.debug("req fetchzones: url='%s', data='%s'" % (self.request_url, urlencode(data) ) )
         response = urlopen(
             Request(
                 "%s?%s" % (self.request_url, urlencode(data)),
                 headers=self.headers
             )
         )
         response_data = json.load(response)['cpanelresult']
-        logger.debug(response_data)
+        logger.debug("rsp fetchzones: data='%s'" % json.dumps(response_data, indent=4))
         matching_zones = {zone for zone in response_data['data'][0]['zones'] if record_domain == zone or record_domain.endswith('.' + zone)}
         if matching_zones:
-            cpanel_zone = max(matching_zones, key = len)
+            cpanel_zone = min(matching_zones, key = len)
```

Comment:
Why use min instead of max? We want to use the zone with the longest matching name to avoid issue with zone delegation.

Reply:
it did not work for me trying to get a certificate for a hosted subdomain, the problem is that the subdomain appears on the list of zones - however I just noticed those hosted subdomain records are empty so I should have just filtered the empty ones out instead of taking min

Reply:
as for the token support, think it comes from another pull request which you already merged with

Reply:
i will create another pull request - based on latest version